### ISSUE TITLE
Azure SD Managed Identity auth doesn't work inside Azure Container Apps

### ISSUE THREAD

**Issue by MattiasAng** [VictoriaMetrics/VictoriaMetrics#3830]
### Describe the bug

When running VictoriaMetrics inside a [Azure Container App](https://learn.microsoft.com/en-us/azure/container-apps/overview) and using Azure service discovery with Managed Identity authentication method, I am unable to get a token.

### To Reproduce

Create a Container App and setup a VMAgent with Azure SD configuration and `ManagedIdentity` as auth method.
```
scrape_configs:
- job_name: test
  azure_sd_configs:
    - subscription_id: REDACTED
      authentication_method: 'ManagedIdentity'
  relabel_configs:
    - source_labels: [instance]
      regex: (.+)
      replacement: ${1}:1000
      target_label: __address__
```

Authentication will keep failing due to an empty token received. The actual result when using an older API version is either 401 or a response that older API versions doesn't work.
``

### Version

`vmagent-20230209-232200-tags-v1.87.1-0-g291c4197`

### Logs

`2023-02-16T16:58:04.476571186Z 2023-02-16T16:58:04.476Z error   VictoriaMetrics/lib/promscrape/config.go:1044   skipping azure_sd_config targets for job_name "test" because of error: cannot list virtual machines: cannot execute azure api request at /subscriptions/REDACTED/providers/Microsoft.Compute/virtualMachines?api-version=2022-03-01: unexpected status code returned from "https://management.azure.com/subscriptions/REDACTED/providers/Microsoft.Compute/virtualMachines?api-version=2022-03-01": 401; expecting 200; response body: "{\"error\":{\"code\":\"AuthenticationFailedMissingToken\",\"message\":\"Authentication failed. The 'Authorization' header is missing the access token.\"}}"`

**Comment by valyala** [comment]
Re-opening the issue until [the fix](6d019a3c371861f42e5bee70101c949d8b782bbf) is included in the next release of VictoriaMetrics and vmagent.

**Comment by valyala** [comment]
[The fix](https://github.com/VictoriaMetrics/VictoriaMetrics/commit/6d019a3c371861f42e5bee70101c949d8b782bbf) has been included in [the 1.87.2 release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the issue as fixed then.

**Comment by valyala** [comment]
FYI, the fix has been included also into [v1.88.0](https://docs.victoriametrics.com/CHANGELOG.html) and all the newer releases.
